### PR TITLE
fzf: patch `bc` dependency in fzf-tmux

### LINKED ISSUE #225579: fzf 0.39.0 missing runtime dependency bc
### Describe the bug
Installing and using fzf breaks with the following error message:

```
.nix-profile/bin/fzf-tmux: line 183: bc: command not found
```

### Steps To Reproduce
1. Install fzf with tmux integration, I did it via home manager modules
2. Call fzf with tmux integration within tmux

### Expected behavior
The tmux pop up appears with history to filter

### Notify maintainers

@Br1ght0ne @Ma27 @zowoq

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
 - system: `"x86_64-linux"`
 - host os: `Linux 5.15.105, NixOS, 23.05 (Stoat), 23.05pre470969.0e19daa510e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.13.3`
 - channels(root): `"nixos, nixos-hardware, phps"`
 - channels(daniels): `"home-manager, nixpkgs, phps"`
 - nixpkgs: `/home/daniels/.nix-defexpr/channels/nixpkgs`
```

Looks like this got introduced with https://github.com/NixOS/nixpkgs/pull/224377. The runtime dependency was added with that new version: https://github.com/junegunn/fzf/compare/0.38.0...0.39.0#diff-29747b0804ea04e1d230bd8651260a6b8687374cc1ae4c2d70ec26f05c4f719bR183 Line 183 of file `bin/fzf-tmux` I wasn't able to find a proper URL to link to that line within the diff.

### PR DESCRIPTION
**Pull request by msfjarvis**
###### Description of changes

Fixes #225579

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by DanielSiepmann** [comment]
Worked for me

**Comment by msfjarvis** [comment]
Rebased on current master.

**Review by Ma27 — APPROVED** [review]

File: pkgs/tools/misc/fzf/default.nix [hunk 1]
```diff
@@ -64,6 +66,7 @@ buildGoModule rec {
 
   postInstall = ''
     install bin/fzf-tmux $out/bin
+    wrapProgram $out/bin/fzf-tmux --prefix PATH : "${lib.makeBinPath [ bc ]}"
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/fzf/default.nix:69]: bc is just used here https://github.com/junegunn/fzf/blob/master/bin/fzf-tmux#L183 so patching would probably be a better idea

**Comment by SuperSandro2000** [comment]
also maybe we should use reshole for those scripts

File: pkgs/tools/misc/fzf/default.nix (hunk 1 above)
Comment (msfjarvis) [reply pkgs/tools/misc/fzf/default.nix]: Replaced it with a `substituteInPlace`

```diff
--- /home/msfjarvis/.nix-profile/bin/fzf-tmux	1970-01-01 05:30:01.000000000 +0530
+++ ./result/bin/fzf-tmux	1970-01-01 05:30:01.000000000 +0530
@@ -180,7 +180,7 @@
 envs="export TERM=$TERM "
 if [[ "$opt" =~ "-E" ]]; then
   tmux_version=$(tmux -V | sed 's/[^0-9.]//g')
-  if [[ $(awk '{print ($1 > 3.2)}' <<< "$tmux_version" 2> /dev/null || bc -l <<< "$tmux_version > 3.2") = 1 ]]; then
+  if [[ $(awk '{print ($1 > 3.2)}' <<< "$tmux_version" 2> /dev/null || /nix/store/fqba9jy3d6f2p67qhdmbdfmvpmrvyx1x-bc-1.07.1/bin/bc -l <<< "$tmux_version > 3.2") = 1 ]]; then
     FZF_DEFAULT_OPTS="--border $FZF_DEFAULT_OPTS"
     opt="-B $opt"
   elif [[ $tmux_version = 3.2 ]]; then
```

**Pull request merged**